src/quick-lint-js/fe/parse.cpp

```diff
@@ -402,6 +402,80 @@ void parser::error_on_invalid_as_const(expression* ast,
   }
 }
 
+void parser::error_on_pointless_compare_against_literal(
+    expression::binary_operator* ast) {
+  auto is_comparison_operator = [](string8_view s) -> bool {
+    return s == u8"=="sv || s == u8"==="sv || s == u8"!="sv || s == u8"!=="sv;
+  };
+  auto is_strict_operator = [](string8_view op) -> bool {
+    return op == u8"==="sv || op == u8"!=="sv;
+  };
+  auto is_regexp_literal = [](expression::literal* reg_exp) -> bool {
+    return !(reg_exp->span().begin() == reg_exp->span().end() ||
+             *reg_exp->span().begin() != u8'/');
+  };
+  auto get_condition_result = [](string8_view equals_operator) -> string8_view {
+    return (equals_operator == u8"==="sv || equals_operator == u8"=="sv)
+               ? u8"false"sv
+               : u8"true"sv;
+  };
+
+  for (int i = 0; i < ast->child_count() - 1; i++) {
+    source_code_span op_span = ast->operator_spans_[i];
+    if (!is_comparison_operator(op_span.string_view())) {
+      continue;
+    }
+
+    for (expression* child : {ast->child(i), ast->child(i + 1)}) {
+      string8_view condition = get_condition_result(op_span.string_view());
+
+      switch (child->without_paren()->kind()) {
+      case expression_kind::_class:
+        this->diag_reporter_->report(diag_pointless_comp_against_class_literal{
+            .equals_operator = op_span, .condition = condition});
+        goto EXIT_LOOP;
+        break;
+      case expression_kind::array:
+        if (is_strict_operator(op_span.string_view())) {
+          if (child->without_paren()->span().string_view() == u8"[]"sv) {
```

Comment:
Must fix: This doesn't work if the array literal has a space or comment inside. Instead of comparing strings, query if the array is any children.
```suggestion
          if (child->without_paren()->child_count() == 0) {
```